### PR TITLE
Implement JIT support for flattened value type arrays

### PR DESCRIPTION
**Pull request by Leonardo2718**
For now, the implementation generates VM helpers calls for the
aaload and aastore operations.

Signed-off-by: Leonardo Banderali <leonardo2718@protonmail.com>

### CODE REVIEW

**Comment by 0xdaryl** [comment]
Jenkins test sanity,extended xlinuxval jdknext

**Comment by 0xdaryl** [comment]
@a7ehuo , would you mind reviewing these changes as well please?

File: runtime/compiler/ilgen/Walker.cpp [hunk 1]
```diff
@@ -6079,6 +6079,22 @@ TR_J9ByteCodeIlGenerator::loadFromCallSiteTable(int32_t callSiteIndex)
 void
 TR_J9ByteCodeIlGenerator::loadArrayElement(TR::DataType dataType, TR::ILOpCodes nodeop, bool checks)
    {
+   if (TR::Compiler->om.areValueTypesEnabled() && dataType == TR::Address)
```
Comment (a7ehuo) [inline runtime/compiler/ilgen/Walker.cpp:6082]: How come we don't check if the array element is value type first and then if it's flattened before going down this path to call VM helper to load flattened array element?

Comment (Leonardo2718) [reply runtime/compiler/ilgen/Walker.cpp:6082]: `aaload` and `aastore` don't use the constant pool, so the type of array elements is not known (at least, not without some dataflow analysis). The only thing that's known is that it's an array of some object type.

**Comment by a7ehuo** [comment]
LGTM

**Review by 0xdaryl — APPROVED** [review]

**Pull request merged**